src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java

```diff
@@ -69,33 +70,72 @@ private void logResults(final Report releaseReport) {
         prepareRepositoryForRelease(repository);
-        report.merge(releaseOnPlatforms(repository, platforms));
-        cleanRepositoryAfterRelease(repository, report);
+        final Report report = releaseOnPlatforms(repository, platforms, released);
+        cleanRepositoryAfterRelease(repository, report.hasFailures());
         return report;
     }
 
-    private Report releaseOnPlatforms(final Repository repository, final List<PlatformName> platforms) {
+    private Report releaseOnPlatforms(final Repository repository, final List<PlatformName> platforms,
+            final Set<PlatformName> released) {
         final Report report = Report.releaseReport();
         for (final PlatformName platformName : platforms) {
-            LOGGER.info(() -> "Releasing on " + platformName + " platform.");
-            try {
-                this.getReleaseMaker(platformName).makeRelease(repository);
-                report.addResult(ReleaseResult.successfulRelease(platformName));
-            } catch (final ReleaseException exception) {
-                report.addResult(ReleaseResult.failedRelease(platformName, ExceptionUtils.getStackTrace(exception)));
-                break;
+            if (released.contains(platformName)) {
+                LOGGER.info(
+                        () -> "Skipping " + platformName + " platform, the release has been already performed there.");
+            } else {
+                final Report platformReport = releaseOnPlatform(repository, platformName);
+                report.merge(platformReport);
+                if (platformReport.hasFailures()) {
+                    break;
+                }
             }
         }
         return report;
     }
 
+    private Report releaseOnPlatform(final Repository repository, final PlatformName platformName) {
+        final var report = Report.releaseReport();
+        try {
+            LOGGER.info(() -> "Releasing on " + platformName + " platform.");
+            getReleaseMaker(platformName).makeRelease(repository);
+            saveProgress(platformName, repository);
+            report.addResult(ReleaseResult.successfulRelease(platformName));
+        } catch (final Exception exception) {
+            report.addResult(ReleaseResult.failedRelease(platformName, ExceptionUtils.getStackTrace(exception)));
+        }
+        return report;
+    }
+
+    private void saveProgress(final PlatformName platformName, final Repository repository) {
+        this.releaseState.saveProgress(repository.getName(), repository.getVersion(), platformName);
+    }
+
     private void prepareRepositoryForRelease(final Repository repository) {
         this.releaseManager.prepareForRelease(repository);
     }
 
-    private void cleanRepositoryAfterRelease(final Repository repository, final Report report) {
-        if (!report.hasFailures()) {
+    private void cleanRepositoryAfterRelease(final Repository repository, final boolean hasFailures) {
```

Comment:
I find this name confusing, as it does not match always to what the method does (i.e. sometimes the repo is clean but some others not). I would suggest to either rename the or the move the if clause outside